<launch>
	<node pkg="tf" type="static_transform_publisher" name="tf_base2mid360" args="-0.425 0 1.005 0 0 0 /base_link /mid360_frame 10" />
	<node pkg="tf" type="static_transform_publisher" name="tf_mid3602velodyne" args="0 0 -0.075 0 0 0 /mid360_frame /velodyne_frame 10" />
	<node pkg="tf" type="static_transform_publisher" name="tf_base2laser" args="0.075 0 0.49 0 0 0 /base_link /laser 10" />
	<node pkg="tf" type="static_transform_publisher" name="tf_base2mid360_new" args="-0.425 0 1.005 0 0 0 /base_link /mid360_frame_new 10" />


	<!--node pkg="tsukuba_challenge2021" name="livox_tf" type="livox_tf.py"/-->

	<node pkg="ypspur_ros" type="ypspur_ros" name="ypspur_ros" respawn="true">
		<param name="port" value="/dev/ttyACM_YP" />
		<param name="param_file" value="/home/nakaba/icart-middle.param" />
		<param name="publish_odom_tf" value="false" />
	</node>


	<node pkg="urg_node" type="urg_node" name="urg_node" respawn="true">
		<param name="serial_port" value="/dev/ttyACM_URG" />
		<param name="angle_min" value="-2" />
		<param name="angle_max" value="2" />
		<remap from="scan" to="scan_livox_front_low_move"/>

	</node>

	<!-- MID360 -->
	<!--user configure parameters for ros start-->
  <group ns="mid360">
		<arg name="lvx_file_path" default="livox_test.lvx"/>
		<arg name="bd_list" default="100000000000000"/>
		<arg name="xfer_format" default="0"/>
		<arg name="multi_topic" default="0"/>
		<arg name="data_src" default="0"/>
		<arg name="publish_freq" default="10.0"/>
		<arg name="output_type" default="0"/>
		<arg name="rviz_enable" default="false"/>
		<arg name="rosbag_enable" default="false"/>
		<arg name="cmdline_arg" default="$(arg bd_list)"/>
		<arg name="msg_frame_id" default="mid360_frame"/>
		<arg name="lidar_bag" default="true"/>
		<arg name="imu_bag" default="true"/>
		<!--user configure parameters for ros end-->

		<param name="xfer_format" value="$(arg xfer_format)"/>
		<param name="multi_topic" value="$(arg multi_topic)"/>
		<param name="data_src" value="$(arg data_src)"/>
		<param name="publish_freq" type="double" value="$(arg publish_freq)"/>
		<param name="output_data_type" value="$(arg output_type)"/>
		<param name="cmdline_str" type="string" value="$(arg bd_list)"/>
		<param name="cmdline_file_path" type="string" value="$(arg lvx_file_path)"/>
		<param name="user_config_path" type="string" value="$(find livox_ros_driver2)/config/MID360_config.json"/>
		<param name="frame_id" type="string" value="$(arg msg_frame_id)"/>
		<param name="enable_lidar_bag" type="bool" value="$(arg lidar_bag)"/>
		<param name="enable_imu_bag" type="bool" value="$(arg imu_bag)"/>

		<node name="livox_lidar_publisher2" pkg="livox_ros_driver2"
					type="livox_ros_driver2_node" required="true"
					output="screen" args="$(arg cmdline_arg)"/>
  </group>

	<!-- usb_acmノードを実行する -->
	<node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
			<!-- カメラの指定　defaultは/dev/video0 -->
			<param name="video_device" value="/dev/video0"/>
			<!-- 画像の大きさ（横幅）の指定 -->
			<param name="image_width" value="640"/>
			<!-- 画像の大きさ（縦幅）の指定 -->
			<param name="image_height" value="480"/>
			<!-- 画像フォーマット設定　"mjpeg" or "yuyv" or "uyvy" -->
			<param name="pixel_format" value="mjpeg"/>
			<!-- fpsの指定　frame per second -->
			<param name="framerate" value="5"/>
	</node>



	<!-- declare arguments with default values -->
	<arg name="calibration" default="$(find velodyne_pointcloud)/params/VLP16db.yaml"/>
	<arg name="device_ip" default="" />
	<arg name="frame_id" default="velodyne_frame" />
	<arg name="manager" default="$(arg frame_id)_nodelet_manager" />
	<arg name="max_range" default="130.0" />
	<arg name="min_range" default="0.4" />
	<arg name="pcap" default="" />
	<arg name="port" default="2368" />
	<arg name="read_fast" default="false" />
	<arg name="read_once" default="false" />
	<arg name="repeat_delay" default="0.0" />
	<arg name="rpm" default="600.0" />
	<arg name="gps_time" default="false" />
	<arg name="pcap_time" default="false" />
	<arg name="cut_angle" default="-0.01" />
	<arg name="timestamp_first_packet" default="false" />
	<arg name="laserscan_ring" default="-1" />
	<arg name="laserscan_resolution" default="0.007" />
	<arg name="organize_cloud" default="false" />

	<!-- start nodelet manager and driver nodelets -->
	<include file="$(find velodyne_driver)/launch/nodelet_manager.launch">
		<arg name="device_ip" value="$(arg device_ip)"/>
		<arg name="frame_id" value="$(arg frame_id)"/>
		<arg name="manager" value="$(arg manager)" />
		<arg name="model" value="VLP16"/>
		<arg name="pcap" value="$(arg pcap)"/>
		<arg name="port" value="$(arg port)"/>
		<arg name="read_fast" value="$(arg read_fast)"/>
		<arg name="read_once" value="$(arg read_once)"/>
		<arg name="repeat_delay" value="$(arg repeat_delay)"/>
		<arg name="rpm" value="$(arg rpm)"/>
		<arg name="gps_time" value="$(arg gps_time)"/>
		<arg name="pcap_time" value="$(arg pcap_time)"/>
		<arg name="cut_angle" value="$(arg cut_angle)"/>
		<arg name="timestamp_first_packet" value="$(arg timestamp_first_packet)"/>
	</include>

	<!-- start transform nodelet -->
	<include file="$(find velodyne_pointcloud)/launch/transform_nodelet.launch">
		<arg name="model" value="VLP16"/>
		<arg name="calibration" value="$(arg calibration)"/>
		<arg name="manager" value="$(arg manager)" />
		<arg name="fixed_frame" value="" />
		<arg name="target_frame" value="" />
		<arg name="max_range" value="$(arg max_range)"/>

		<arg name="min_range" value="$(arg min_range)"/>
		<arg name="organize_cloud" value="$(arg organize_cloud)"/>
	</include>

	<!-- start laserscan nodelet -->
	<include file="$(find velodyne_pointcloud)/launch/laserscan_nodelet.launch">
		<arg name="manager" value="$(arg manager)" />
		<arg name="ring" value="$(arg laserscan_ring)"/>
		<arg name="resolution" value="$(arg laserscan_resolution)"/>
	</include>

	<node pkg="tc2025" name="flag_maker" type="flag_maker.py" output="screen" >
	</node>

	<node pkg="tc2025" name="waypoint_gui" type="waypoint_gui.py" output="screen" >
	</node>


</launch>
